sr_port/op_indq2.c

```diff
@@ -0,0 +1,82 @@
+/****************************************************************
+ *								*
+ * Copyright 2001, 2012 Fidelity Information Services, Inc	*
+ *								*
+ * Copyright (c) 2017 YottaDB LLC. and/or its subsidiaries.	*
+ * All rights reserved.						*
+ *								*
+ *	This source code contains the intellectual property	*
+ *	of its copyright holder(s), and is made available	*
+ *	under a license.  If you do not know the terms of	*
+ *	the license, please stop and do not read further.	*
+ *								*
+ ****************************************************************/
+
+#include "mdef.h"
+
+#include "compiler.h"
+#include "opcode.h"
+#include "op.h"
+#include "lv_val.h"
+#include "mvalconv.h"
+#include "glvn_pool.h"
+#include "callg.h"
+
+error_def(ERR_QUERY2);
+error_def(ERR_VAREXPECTED);
+
+GBLREF	symval			*curr_symval;
+
+void	op_indq2(mval *dst, uint4 indx, mval *direct)
+{
+	int		i;
+	glvn_pool_entry	*slot;
+	int4		dummy_intval;
+	intszofptr_t	n;
+	lv_val		*lv;
+	mval		*key;
+	opctype		oc;
+	gparam_list	paramlist;
+	ht_ent_mname	*tabent;
+	mname_entry	targ_key;
+	lvname_info	*lvn_info;
+	DCL_THREADGBL_ACCESS;
+
+	SETUP_THREADGBL_ACCESS;
+	MV_FORCE_NUM(direct);
+	if (!MV_IS_TRUEINT(direct, &dummy_intval) || (direct->m[1] != (1 * MV_BIAS) && direct->m[1] != (-1 * MV_BIAS)))
+		rts_error(VARLSTCNT(1) ERR_QUERY2);
+	slot = &((TREF(glvn_pool_ptr))->slot[indx]);
+	oc = slot->sav_opcode;
+	if (OC_SAVLVN == oc)
+	{	/* lvn */
+		lvn_info = (lvname_info *)&slot->glvn_info;
+		n = lvn_info->total_lv_subs + 2;
+		assert(3 <= n);
+		/* Prepare paramlist to pass to "op_fnquery" or "op_fnreversequery" */
+		paramlist.n = n;
+		paramlist.arg[0] = dst;
+		paramlist.arg[1] = slot->lvname;
+		targ_key.var_name = slot->lvname->str;
+		COMPUTE_HASH_MNAME(&targ_key);
+		targ_key.marked = FALSE;
+		if (add_hashtab_mname_symval(&curr_symval->h_symtab, &targ_key, NULL, &tabent))
+			lv_newname(tabent, curr_symval);
+		paramlist.arg[2] = (lv_val *)tabent->value;
+		for (i = 3; i < n; i++)
+			paramlist.arg[i] = lvn_info->lv_subs[i-3];
```

Comment:
Spaces should surround minus sign in [..].

Reply:
Done.